Tasks/JenkinsQueueJobV3/jenkinsqueuejobtask.ts

```diff
@@ -0,0 +1,120 @@
+        }
+
+        this.captureConsole = tl.getBoolInput('captureConsole', true);
+        // capturePipeline is only possible if captureConsole mode is enabled
+        this.capturePipeline = this.captureConsole ? tl.getBoolInput('capturePipeline', true) : false;
+
+        this.pollIntervalMillis = 5000; // five seconds is what the Jenkins Web UI uses
+
+        this.parameterizedJob = tl.getBoolInput('parameterizedJob', true);
+        // jobParameters are only possible if parameterizedJob is enabled
+        this.jobParameters = this.parameterizedJob ? tl.getDelimitedInput('jobParameters', '\n', false) : [];
+
+        this.jobQueueUrl = util.addUrlSegment(this.serverEndpointUrl, util.convertJobName(this.jobName)) + ((this.parameterizedJob) ? '/buildWithParameters?delay=0sec' : '/build?delay=0sec');
+        tl.debug('jobQueueUrl=' + this.jobQueueUrl);
+        this.teamJobQueueUrl = util.addUrlSegment(this.serverEndpointUrl, '/team-build/' + ((this.parameterizedJob) ? 'buildWithParameters/' : 'build/') + this.jobName + '?delay=0sec');
+        tl.debug('teamJobQueueUrl=' + this.teamJobQueueUrl);
+        this.teamPluginUrl = util.addUrlSegment(this.serverEndpointUrl, '/pluginManager/available');
+        tl.debug('teamPluginUrl=' + this.teamPluginUrl);
+
+        this.teamBuildPluginAvailable = false;
+        // 'Build.StagingDirectory' is available during build.
+        // It is kept here (different than what is used during release) to maintain
+        // compatibility with other tasks relying on Jenkins results being placed in this folder.
+        let resultsDirectory: string = tl.getVariable('Build.StagingDirectory');
+        if (!resultsDirectory) {
+            // 'System.DefaultWorkingDirectory' is available during build and release
+            resultsDirectory = tl.getVariable('System.DefaultWorkingDirectory');
+        }
+        this.saveResultsTo = path.join(resultsDirectory, 'jenkinsResults');
+
+        this.strictSSL = ('true' !== tl.getEndpointDataParameter(this.serverEndpoint, 'acceptUntrustedCerts', true));
+        tl.debug('strictSSL=' + this.strictSSL);
+
+        this.NO_CRUMB = 'NO_CRUMB';
+        this.crumb = this.NO_CRUMB;
+
+        this.shouldFail = false;
+        this.failureMsg = '';
+    }
+}
+
+async function doWork() {
+    try {
+        tl.setResourcePath(path.join( __dirname, 'task.json'));
+
+        const taskOptions: TaskOptions = new TaskOptions();
+        const jobQueue: JobQueue = new JobQueue(taskOptions);
+        const queueUri = await util.pollSubmitJob(taskOptions);
+        console.log(tl.loc('JenkinsJobQueued'));
+        const rootJob = await util.pollCreateRootJob(queueUri, jobQueue, taskOptions);
+        //start the job queue
+        jobQueue.Start();
+        //store the job name in the output variable
+        tl.setVariable('JENKINS_JOB_ID', rootJob.ExecutableNumber.toString());
+    } catch (e) {
+        tl.debug(e.message);
```

Comment:
I'm not sure this line is necessary if you're going to log the error and then set the result with this value.